File: src/main/java/nl/crosscode/x509certbuddy/actions/RemoveAllCertsAction.java
package nl.crosscode.x509certbuddy.actions;

import com.intellij.icons.AllIcons;
import com.intellij.openapi.actionSystem.AnAction;
import com.intellij.openapi.actionSystem.AnActionEvent;
import nl.crosscode.x509certbuddy.x509CertAssistantFactory;
import org.jetbrains.annotations.NotNull;

public class RemoveAllCertsAction extends AnAction {

    public RemoveAllCertsAction() {
        this.getTemplatePresentation().setIcon(AllIcons.Actions.Cancel);
        this.getTemplatePresentation().setText("Remove All Certificates");
    }

    @Override
    public void actionPerformed(@NotNull AnActionEvent e) {
        x509CertAssistantFactory.getX509CertAssistant(e.getProject()).removeAllCerts(null);
    }
}

File: src/main/kotlin/nl/crosscode/x509certbuddy/actions/RemoveAllCertsAction.kt
package nl.crosscode.x509certbuddy.actions

import com.intellij.icons.AllIcons
import com.intellij.openapi.actionSystem.AnAction
import com.intellij.openapi.actionSystem.AnActionEvent
import nl.crosscode.x509certbuddy.X509CertAssistantFactory

class RemoveAllCertsAction : AnAction() {
    init {
        this.templatePresentation.icon = AllIcons.Actions.GC
        this.templatePresentation.text = "Remove All Certificates"
    }

    override fun actionPerformed(e: AnActionEvent) {
        if (e.project == null) {
            e.presentation.isEnabled = false
            return
        }
        val assistant = X509CertAssistantFactory.getInstance(e.project!!)
        assistant?.removeAllCerts()
    }

    override fun update(e: AnActionEvent) {
        if (e.project == null) {
            e.presentation.isEnabled = false
            return
        }
        val assistant = X509CertAssistantFactory.getInstance(e.project!!)
        e.presentation.isEnabled = assistant?.hasCerts() ?: false
        super.update(e)
    }
}
